app/routes/data-module.js

```diff
@@ -22,10 +22,16 @@ export default Route.extend({
       moduleName = '@' + moduleName;
     }
     let mappingInfo = this.legacyModuleMappings.getNewModuleFromOld(moduleName, mappings);
-    return this.transitionTo(`project-version.modules.module`,
-      'ember-data',
-      'release',
-      mappingInfo.module);
+    if(mappingInfo.module === '@ember-data') {
+      return this.transitionTo(`project-version`,
+        'ember-data',
+        'release');
+    } else {
+      return this.transitionTo(`project-version.modules.module`,
+        'ember-data',
+        'release',
+        mappingInfo.module);
+    }
```

Comment:
I made an early exit and updated the code style for readability.

```suggestion

    if (mappingInfo.module === '@ember-data') {
      return this.transitionTo(
        'project-version',
        'ember-data',
        'release'
      );
    }
    
    return this.transitionTo(
      'project-version.modules.module',
      'ember-data',
      'release',
      mappingInfo.module
    );
```